Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js

```diff
@@ -0,0 +1,153 @@
+import { getRequiredElement } from "../Blazorise/utilities.js?v=1.2.3.0";
+import "./vendors/sigpad.js?v=1.2.3.0";
+
+const _instances = [];
+
+export function initialize(dotNetAdapter, element, elementId, options) {
+    element = getRequiredElement(element, elementId);
+
+    if (!element)
+        return;
+
+    const sigpad = new SignaturePad(element, {
+        minWidth: options.minLineWidth || 0.5,
+        maxWidth: options.maxLineWidth || 2.5,
+        backgroundColor: options.backgroundColor || "rgba(0,0,0,0)",
+        penColor: options.penColor || "black",
+        velocityFilterWeight: options.velocityFilterWeight || 0.7,
+        dotSize: options.dotSize,
+        throttle: options.throttle || 16,
+        minPointDistance: options.minPointDistance || 0.5,
+    });
+
+    if (options.value) {
+        sigpad.fromData(options.value, { clear: false });
+    }
+
+    if (options.readOnly === true) {
+        sigpad.off();
+    }
+
+    const instance = {
+        options: options,
+        sigpad: sigpad,
+    };
+
+    registerToEvents(dotNetAdapter, instance);
+
+    _instances[elementId] = instance;
+}
+
+export function destroy(element, elementId) {
+    const instances = _instances || {};
+    const instance = instances[elementId];
+
+    if (instance) {
+        if (instance.sigpad) {
+            instance.sigpad.off();
```

Comment:
do we need to call .`off` when destroying? Isn't `off` to set `readonly`? We don't usually have the approach of disabling components when initiating the disposal process. Does `sigpad `have any api to actually destroy?

Reply:
The `.off()` is the only method that is closes to destroy on sigpad.

Reply:
I'm confused is off supposed to set the `sigpad` to `readonly` or to destroy it? Are those not two different things?

Reply:
To some kind of read-only state, as far as I'm aware. https://github.com/szimek/signature_pad#api

> // Unbinds all event handlers
> signaturePad.off();